action.yml

```diff
@@ -3,8 +3,35 @@ author: 'ApeWorX Ltd.'
 description: 'Run an ape command'
 
 runs:
-  using: 'docker'
+  using: 'composite'
   # TODO: Eventually use `stable`
-  image: 'docker://apeworx/ape:latest'
-  args:
-  - 'test'
+  steps:
+  
+  - uses: actions/cache@v2
+    with:
+      path: |
+        /home/runner/.solcx
+        /home/runner/.vvm/vyper-*
+      key: ${{ runner.os }}-compiler-cache
+
+  - uses: actions/cache@v2
+    with:
+      path: |
+        ${{ github.workspace }}/.build
+      key: ${{ runner.os }}-build-cache
+
+  # todo: select version of python via input
+  - uses: actions/setup-python@v2
+    with: 
+      python-version: 3.8
+  
+  # todo: cache pip
+  # - uses: actions/cache@v2
+
+  # todo: select version of ape via input  
+  - run: pip install eth-ape
+    shell: bash
+
+  # todo: allow selecting plugins
+  - run: ape plugins install solidity vyper
```

Comment:
Can we use the project's config file to determine which compilers are needed before installing them?

Reply:
we just resolved this :)